checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -1091,10 +1089,12 @@ private boolean isTreeSymbolEffectivelyFinalOrUnmodifiable(Tree tree) {
 
     @Override
     public Void visitIdentifier(IdentifierTree tree, Void p) {
-        Node node = atypeFactory.getNodeForTree(tree);
-        if (node instanceof FieldAccessNode) {
-            Node receiverNode = ((FieldAccessNode) node).getReceiver();
-            if (receiverNode instanceof ImplicitThisLiteralNode) {
+        if (TreeUtils.isFieldAccess(tree)) {
+            Tree parent = getCurrentPath().getParentPath().getLeaf();
+            if (parent.getKind() != Kind.MEMBER_SELECT
+                    || ((MemberSelectTree) parent).getExpression() == tree) {
```

Comment:
Instead of `getExpression` shouldn't this be `getIdentifier()` and then some call on `tree` to get its `Name`?
I'm not quite understanding the logic otherwise and a comment would be helpful.